src/compas_robots/model/joint.py

```diff
@@ -100,11 +93,9 @@ def data(self):
             "reference_position": self.reference_position,
         }
 
-    @data.setter
-    def data(self, data):
-        self.rising = data["rising"]
-        self.falling = data["falling"]
-        self.reference_position = data["reference_position"]
+    @classmethod
+    def from_data(cls, data):
+        return cls(data["rising"], data["falling"], data["reference_position"])
```

Comment:
I think here it's safe to omit `from_data` as the `data` dict matches signature of `__init__()`

Reply:
True, removed all redundant `from_data` in commit https://github.com/compas-dev/compas_robots/pull/1/commits/db0b3d35685a384ac9c30e493add6ba8e1e9945b